janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java

```diff
@@ -14,24 +14,52 @@
 
 package org.janusgraph.graphdb.relations;
 
+import org.apache.commons.lang3.StringUtils;
 import org.janusgraph.util.encoding.LongEncoding;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
 import java.util.Arrays;
 
+import static org.janusgraph.util.encoding.LongEncoding.STRING_ENCODING_MARKER;
+
 /**
  * @author Matthias Broecheler (me@matthiasb.com)
  */
 
 public final class RelationIdentifier implements Serializable {
 
-    public static final String TOSTRING_DELIMITER = "-";
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelationIdentifier.class);
+    public static final String JANUSGRAPH_RELATION_DELIMITER = "JANUSGRAPH_RELATION_DELIMITER";
+    public static final String TOSTRING_DELIMITER;
```

Comment:
It indeed must have only one character (otherwise `id.split(TOSTRING_DELIMITER)` could lead to wrong results). We have a check for that in the static initializer of this class. The variable is a String for convenience reasons - no need to cast string to char when we load from env variable (or system properties), plus no need to cast char back to string when we do `id.split(TOSTRING_DELIMITER);`.

Environment variables and System properties are always of String type, so there shouldn't be ambiguity/confusion.

Reply:
I'll explain in the doc that it should be a string with a single character.